packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/decorators/CSSProps.kt

```diff
@@ -0,0 +1,174 @@
+      color
+    )
+  }
+}
+
+inline fun <reified T : View> ViewDefinitionBuilder<T>.UseBorderWidthProps(crossinline body: (view: T, edge: LogicalEdge, width: Float?) -> Unit) {
+  PropGroup(
+    ViewProps.BORDER_WIDTH,
+    ViewProps.BORDER_LEFT_WIDTH,
+    ViewProps.BORDER_RIGHT_WIDTH,
+    ViewProps.BORDER_TOP_WIDTH,
+    ViewProps.BORDER_BOTTOM_WIDTH,
+    ViewProps.BORDER_START_WIDTH,
+    ViewProps.BORDER_END_WIDTH
+  ) { view: T, index: Int, width: Float? ->
+    body(view, LogicalEdge.entries[index], width)
+  }
+}
+
+private fun <T : View> ViewDefinitionBuilder<T>.UseBorderWidthProps() {
+  enforceType<ViewDefinitionBuilder<View>>(this)
+  UseBorderWidthProps { view, edge, width ->
+    BackgroundStyleApplicator.setBorderWidth(
+      view,
+      edge,
+      width ?: Float.NaN
+    )
+  }
+}
+
+inline fun <reified T : View> ViewDefinitionBuilder<T>.UseBorderRadiusProps(crossinline body: (view: T, border: BorderRadiusProp, radius: LengthPercentage?) -> Unit) {
+  PropGroup(
+    ViewProps.BORDER_RADIUS,
+    ViewProps.BORDER_TOP_LEFT_RADIUS,
+    ViewProps.BORDER_TOP_RIGHT_RADIUS,
+    ViewProps.BORDER_BOTTOM_RIGHT_RADIUS,
+    ViewProps.BORDER_BOTTOM_LEFT_RADIUS,
+    ViewProps.BORDER_TOP_START_RADIUS,
+    ViewProps.BORDER_TOP_END_RADIUS,
+    ViewProps.BORDER_BOTTOM_START_RADIUS,
+    ViewProps.BORDER_BOTTOM_END_RADIUS,
+    ViewProps.BORDER_END_END_RADIUS,
+    ViewProps.BORDER_END_START_RADIUS,
+    ViewProps.BORDER_START_END_RADIUS,
+    ViewProps.BORDER_START_START_RADIUS
+  ) { view: T, index: Int, radius: Float? ->
+    body(
+      view,
+      BorderRadiusProp.entries[index],
+      radius?.let { LengthPercentage(it, LengthPercentageType.POINT) }
+    )
+  }
+}
+
+private fun <T : View> ViewDefinitionBuilder<T>.UseBorderRadiusProps() {
+  enforceType<ViewDefinitionBuilder<View>>(this)
+  UseBorderRadiusProps { view, border, radius ->
+    BackgroundStyleApplicator.setBorderRadius(
+      view,
+      border,
+      radius
+    )
+  }
+}
+
+inline fun <reified T : View> ViewDefinitionBuilder<T>.UseBorderStyleProp(crossinline body: (view: T, style: BorderStyle?) -> Unit) {
+  Prop("borderStyle") { view: T, style: String? ->
+    val parsedBorderStyle = style?.let { BorderStyle.fromString(style) }
+    body(view, parsedBorderStyle)
+  }
+}
+
+private fun <T : View> ViewDefinitionBuilder<T>.UseBorderStyleProp() {
+  enforceType<ViewDefinitionBuilder<View>>(this)
+  UseBorderStyleProp { view, style ->
+    BackgroundStyleApplicator.setBorderStyle(
+      view,
+      style
+    )
+  }
+}
+
+inline fun <reified T : View> ViewDefinitionBuilder<T>.UseBackgroundProp(crossinline body: (view: T, color: Int?) -> Unit) {
+  Prop(ViewProps.BACKGROUND_COLOR) { view: T, color: Int? ->
+    body(view, color)
+  }
+}
+
+private fun <T : View> ViewDefinitionBuilder<T>.UseBackgroundProp() {
+  enforceType<ViewDefinitionBuilder<View>>(this)
+  UseBackgroundProp { view, color ->
+    BackgroundStyleApplicator.setBackgroundColor(view, color)
```

Comment:
FYI that `BaseViewManager` in RN 0.77 uses this for `backgroundColor` and `boxShadow` out of the box. Hope is to eventually get border rendering into the same place, but it needs some more work around deprecating some existing funky APIs exposed to view managers for codegen'd delegates.